### ISSUE TITLE
HTTP HEC compatible ingester detect body type

### ISSUE THREAD

**Issue by kris-watts-gravwell** [gravwell/gravwell#805]
## What is the feature to be added?

Make the HEC listener try to detect the "type" of data in the body payload

## Why should we add this feature? (Business justification? What problem is the feature trying to solve?)

The HEC "spec" (I loathe to call it that) doesn't really define the types that can be thrown in the main body payload.  As a result some people chuck strings, some chuck proper JSON objects, some chuck binary slices.  As a result we can sometimes escape or fail to escape payloads.

A great example is cribl when throwing O365 logs.  They get the raw JSON, crack out fields and then post to the HEC API with the fields and the raw JSON shipped as a string.  Because we are trying to deal with JSON raw objects we end up getting their JSON payload as a string which is then escaped.  This is sucks.

So basically we have to look at that payload and try to infer what "type" we were sent, if its a string, treat appropriately, if its an actual JSON object, do that, if it's a damn base64 encoded binary payload, handle that!

## How does this feature address the above problem?

The "spec" is weak and everyone just sort of does what they want, we have to get smart about doing magical stuff.

## Any other comments?

Right now we decode into a json.RawMessage, which is the raw bytes we were given, unfortunately that still strips quotes on strings so we might need to modify that into a custom type.

**Comment by kris-watts-gravwell** [comment]
tested with prod load, LGTM
